### DISCUSSION TITLE
Metronome perfomance issue

### DISCUSSION THREAD

**Discussion by Loreos7** [ppy/osu#24404] [Q&A] [answered]
There is fps drops when you time a map with many timing points

https://github.com/ppy/osu/assets/86934170/da60a4a1-156d-4ed4-9458-a7402332ae18

**Comment by peppy** [comment]
Please link the beatmap

**Reply by Loreos7** [reply]
It's my own map without any hitobjects
[Cigarettes After Sex - Sunsetz (Loreos).zip](https://github.com/ppy/osu/files/12206718/Cigarettes.After.Sex.-.Sunsetz.Loreos.zip)

**Reply by Loreos7** [reply]
It works everywhere, you can try these ones instead:
https://osu.ppy.sh/beatmapsets/1277657#osu/2654506
https://osu.ppy.sh/beatmapsets/1011576#osu/2239445

**Reply by peppy** [reply]
I can't reproduce this. Could you please provide log files? Also could you try on legacy opengl renderer to see if it's any better?


https://github.com/ppy/osu/assets/191335/e118dc14-7ade-4709-9c57-2544020cb37e

**Reply by Loreos7** [reply]
Here are logs:
[runtime.log](https://github.com/ppy/osu/files/12208873/runtime.log)
[performance.log](https://github.com/ppy/osu/files/12208874/performance.log)
[network.log](https://github.com/ppy/osu/files/12208875/network.log)
[database.log](https://github.com/ppy/osu/files/12208876/database.log)
[updater.log](https://github.com/ppy/osu/files/12208877/updater.log)

**Reply by peppy** [reply]
Did you try the different renderer?

**Reply by Loreos7** [reply]
OpenGL legacy performs the same:

https://github.com/ppy/osu/assets/86934170/9b5e0ed9-a826-4794-82ab-ace7c6ca76ba

[performance.log](https://github.com/ppy/osu/files/12208907/performance.log)
[runtime.log](https://github.com/ppy/osu/files/12208908/runtime.log)
[network.log](https://github.com/ppy/osu/files/12208909/network.log)
[updater.log](https://github.com/ppy/osu/files/12208910/updater.log)
[database.log](https://github.com/ppy/osu/files/12208911/database.log)

**Reply by Loreos7** [reply]
Yeah, tried both OpenGL, all are the same. GPU drivers are up to date.

![image](https://github.com/ppy/osu/assets/86934170/c8a60094-7ce0-47da-9191-dc19952367c6)

**Comment by Loreos7** [comment, marked as answer]
I'm happy to say that this problem somehow is gone now.

https://github.com/ppy/osu/assets/86934170/10685afd-08c8-42e6-80b6-0223bf73c7ae

**Reply by peppy** [reply]
Likely a result of https://github.com/ppy/osu-framework/pull/6009. 👍
